marker-api.js

```diff
@@ -1,780 +1,780 @@
-            if (err)
-                return res.status(500).json({message: myMongoErrs.get(err.code)});
-
-            if (!data)
-                return res.status(404).json({message: 'No markers were found.'});
-
-            // Build return data (remove private data)
-            returnData = data.map(function (marker) {
-                var mo = marker.obj.toObject();
-
-                // Return distance
-                mo.distance = marker.dis;
-
-                return _.omit(mo, 'user_id', '__v');
-            });
-
-            res.json({
-                message: 'Markers found: ' + data.length,
-                data: returnData
-            });
-        });
-    },
-
-    admin: {
-        markerUnapproved: function (req, res) {
-            latest_marker_unapproved().then((data) => {
-                res.json({message: 'success', data: data});
-            }, (errData) => {
-                res.json({message: 'failure', data: errData});
-            });
-        },
-        markerById: function (req, res) {
-        },
-        setApproval: function (req, res) {
-            var approved = req.body.approved,
-                marker_id = req.body.marker_id;
-
-            // Only tags are editable for now
-            if (approved === undefined) {
-                return res.status(422).json({message: 'No approval status received.'});
-            }
-
-            // Update marker data
-            Marker.findOne({'_id': marker_id}, function (err, marker) {
-                if (err) {
-                    console.log(err);
-                    return res.status(500).json({message: 'An internal error occurred'});
-                }
-
-                if (!marker)
-                    return res.status(404).json({message: 'No marker was found with id ' + req.query.marker_id});
-
-                marker.approved = approved;
-
-                marker.save(function (err) {
-                    if (err) {
-                        res.json({message: 'An internal error occurred'});
-                    }
-
-                    res.status(200).json({
-                        message: marker._id + ' updated successfully',
-                        approved: marker.approved 
-                    });
-                });
-            }); // end update
-        }
-    }
-}
-
-var myMongoErrs = {
-
-    get: function (num) {
-        return myMongoErrs.errsByCode[String(num)];
-    },
-
-    errsByCode: {
-        "17287": "There is a problem with your coordinates. Please make sure all four coordinates are valid floating point numbers"
-    }
-}
+      }
+
+          // Build return data
+          returnData = marker.toObject();
+          returnData = _.omit(returnData, 'user_id', '__v');
+
+          res.status(201).json({ 
+              message: 'New marker save successful',
+              data: returnData
+          });
+      });
+  }
+
+  function handle_marker_data_failure () {
+      console.log('marker data failure');
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/charlieblevins/dukweb/pullRequest?prid=9134160)
```suggestion
      console.log("marker data failure");
```